sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -0,0 +1,758 @@
+        ----------
+        X : array-like, shape=(n_samples, n_features)
+            The input samples.
+
+        Returns
+        -------
+        raw_predictions : array, shape (n_samples * n_trees_per_iteration,)
+            The raw predicted values.
+        """
+        X = check_array(X, dtype=[X_DTYPE, X_BINNED_DTYPE])
+        check_is_fitted(self, '_predictors')
+        if X.shape[1] != self.n_features_:
+            raise ValueError(
+                'X has {} features but this estimator was trained with '
+                '{} features.'.format(X.shape[1], self.n_features_)
+            )
+        is_binned = getattr(self, '_in_fit', False)
+        n_samples = X.shape[0]
+        raw_predictions = np.zeros(
+            shape=(self._n_trees_per_iteration, n_samples),
+            dtype=self._baseline_prediction.dtype
+        )
+        raw_predictions += self._baseline_prediction
+        for predictors_of_ith_iteration in self._predictors:
+            for k, predictor in enumerate(predictors_of_ith_iteration):
+                predict = (predictor.predict_binned if is_binned
+                           else predictor.predict)
+                raw_predictions[k, :] += predict(X)
+
+        return raw_predictions
+
+    @abstractmethod
+    def _get_loss(self):
+        pass
+
+    @abstractmethod
+    def _encode_y(self, y=None):
+        pass
+
+    @property
+    def n_iter_(self):
+        check_is_fitted(self, '_predictors')
+        return len(self._predictors)
+
+
+class HistGradientBoostingRegressor(BaseHistGradientBoosting, RegressorMixin):
+    """Histogram-based Gradient Boosting Regression Tree.
+
+    This estimator is much faster than
+    :class:`GradientBoostingRegressor<sklearn.ensemble.GradientBoostingRegressor>`
+    for big datasets (n_samples >= 10 000). The input data ``X`` is pre-binned
+    into integer-valued bins, which considerably reduces the number of
+    splitting points to consider, and allows the algorithm to leverage
+    integer-based data structures. For small sample sizes,
+    :class:`GradientBoostingRegressor<sklearn.ensemble.GradientBoostingRegressor>`
+    might be preferred since binning may lead to split points that are too
+    approximate in this setting.
+
+    Parameters
+    ----------
+    loss : {'least_squares'}, optional (default='least_squares')
+        The loss function to use in the boosting process.
+    learning_rate : float, optional (default=0.1)
+        The learning rate, also known as *shrinkage*. This is used as a
+        multiplicative factor for the leaves values. Use ``1`` for no
+        shrinkage.
+    max_iter : int, optional (default=100)
+        The maximum number of iterations of the boosting process, i.e. the
+        maximum number of trees.
+    max_leaf_nodes : int or None, optional (default=None)
+        The maximum number of leaves for each tree. If None, there is no
+        maximum limit.
+    max_depth : int or None, optional (default=None)
```

Comment:
@ogrisel @NicolasHug Is it normal that the default is to grow the full tree? I would have expected a shallower option as default.

Reply:
We're following lightgbm defaults here: depth isn't constrained, but max_leaf_nodes is 31.

The doc is wrong, good catch ;)